.circleci/config.yml

```diff
@@ -1,14 +1,198 @@
+---
 version: 2.1
+executors:
+  golang:
+    docker:
+      - image: circleci/golang:1.14.8-node
+  nodejs:
+    docker:
+      - image: circleci/node:10.0.0
 jobs:
-  build:
+  backend-services-checks:
+    executor: golang
+    steps:
+      - checkout
+      - run:
+          name: Run backend service checks
+          command: make backend-services-checks
+          working_directory: ~/project/litmus-portal
+  frontend-services-checks:
+    executor: nodejs
+    steps:
+      - checkout
+      - run:
+          name: Run frontend service checks
+          command: make frontend-services-checks
+          working_directory: ~/project/litmus-portal
+  docker-build:
```

Comment:
Why aren't we using makefile?

Reply:
Makefile is not helping much while debugging. Now every docker build is separated into steps.

Reply:
We can have each stage defined in the makefile and just call them in the circleci something like `make auth-server-image`

Reply:
Not required. it will increase one more layer. Makefile will be required while doing backend and frontend checks. (A group of commands)

Reply:
And it will be difficult to map envs, as circle and makefile have different format of environment

Reply:
circleci- ${}
Makefile- $()

Reply:
I am saying that we should keep the stages in the make file, like it was before. That way when you have to locally build or test the services you can just run `make build` or `make lint`. For local setup make file is really helpful. And as we are already doing it in the current circleci setup we can reuse these functions in the CI. We did not have any problems in the build previously. You can break the whole build into stages like you are doing now but keep the actual logic in the makefile that will reduce redundancy. But if there's a problem with that you can have the build logic in the CI too but let it be there in the makefile also.

Reply:
We can have only build steps, but having push steps doesn't make sense and Makefile will become too ugly if have make step for every docker build.

Also, if you use Makefile to build docker images. It will build for litmuschaos docker username. I think currently no one is using makefile to build docker images. If they are using it will take around 10 mins to build.
I am planning to have a parameterized bash script which will build local docker images, not from Makefile.

Reply:
Makefile can also take parameterized input I manually change the docker repo name and add in the docker creds as env var to build and push to my registry. We can call `make repo=gdsoumya build` that can build the images for my repo.